File: java/hamming/src/main/java/Hamming.java
public class Hamming {
  public static int compute(String first, String second) {
    if (first.length() != second.length()) {
      throw new java.lang.IllegalArgumentException("Both the strings need
                                                    to be the same length");
    }

    char[] firstChars = first.toCharArray();
    char[] secondChars = second.toCharArray();
    int count = 0;

    for(int i = 0; i < firstChars.length; i++) {
      if(firstChars[i] != secondChars[i]) count++;
    }

    return count;
  }
}

File: go/hamming/hamming.go
package hamming

import "errors"

const testVersion = 4

func Distance(s1, s2 string) (count int, err error) {
	r1 := []rune(s1)
	r2 := []rune(s2)

	if len(r1) != len(r2) {
		err = errors.New("The strings need to be the same length")
		return
	}

	for i, c := range r1 {
		if r2[i] != c {
			count++
		}
	}
	return
}
